vue-app/src/pages/GridPage.vue

```diff
@@ -0,0 +1,376 @@
+              :style="{
+                textDecoration: !el.selected ? 'line-through' : 'none',
+              }"
+            >
+              {{ el.aspect }}
+            </grid-config-button>
+          </grid-button-window>
+
+          <grid-button-window title="Grid options">
+            <grid-config-button
+              @GBClicked="
+                () => {
+                  elements = getGeneratedElements();
+                }
+              "
+            >
+              Generate
+            </grid-config-button>
+
+            <grid-config-button
+              @GBClicked="
+                () => {
+                  _cssProps.gridAutoFlow =
+                    _cssProps.gridAutoFlow == 'dense' ? 'row' : 'dense';
+                }
+              "
+            >
+              {{ _cssProps.gridAutoFlow }}
+            </grid-config-button>
+            <grid-config-slider
+              _name="Element size"
+              :_value="itemSize"
+              @GSchange="
+                (e: InputEvent) => {
+                  const target = e.target as HTMLInputElement;
+                  itemSize = e != null ? target.value : '0';
+                  handleSlider();
+                }
+              "
+            >
+            </grid-config-slider>
+            <grid-config-slider
+              _name="Gap size"
+              :_value="gapSize"
+              @GSchange="
+                (e: InputEvent) => {
+                  const target = e.target as HTMLInputElement;
+                  gapSize = e != null ? target.value : '0';
+                  _cssProps.gap = gapSize;
+                }
+              "
+            >
+            </grid-config-slider>
+          </grid-button-window>
+        </div>
+
+        <div class="code-listing-wrapper">
+          <code-listing :cssProps="_cssProps"> </code-listing>
+        </div>
+
+        <grid :cssProps="_cssProps">
+          <div
+            v-for="element in elements"
+            :key="element.id"
+            :class="element.class + ' element'"
+            :style="element.style"
+          >
+            <span>{{ element.id }}</span>
+          </div>
+        </grid>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import Grid from "../components/Grid.vue";
+import GridConfigButton from "../components/UI/GridButton.vue";
+import CodeListing from "../components/CodeListing.vue";
+import GridConfigSlider from "../components/GridConfigSlider.vue";
+import GridButtonWindow from "../components/GridButtonWindow.vue";
+import { CSSProperties, ref } from "vue";
+import { subtractWithSaturation8bit } from "../utils";
+import { colors } from "../static/GridElements";
+
+type GeneratedElement = {
+  id: number;
+  class: string;
+  style: CSSProperties;
+};
+
+const numberOfElements = 20;
+
+const itemSize = ref<string>("100");
+const gapSize = ref<string>("10");
+
+let colorsArray: Array<Array<number>> = [];
+
+const _cssProps = ref<CSSProperties>({
```

Comment:
what's the purpose of this "_" prefix?
reactive is more suitable here

Reply:
I am not using reactive at all, don't like mixing styles and I won't change it